~~@pomegranited whenever possible, please add my SSH key to CircleCI in order to be able to run the test build. In addition, I am currently getting the following error:~~
```
======================================================================
ERROR: test_launch (tests.integration.test_skytap.TestSkytap)
Test that successful launch opens sharing portal URL in new tab.
----------------------------------------------------------------------
Traceback (most recent call last):
  File "/home/nizarmah/Documents/Work/opencraft/xblock/xblock-skytap/.tox/py38-django30/lib/python3.8/site-packages/xblockutils/base_test.py", line 45, in setUp
    super(SeleniumXBlockTest, self).setUp()
  File "/home/nizarmah/Documents/Work/opencraft/xblock/xblock-skytap/.tox/py38-django30/src/xblock-sdk/workbench/test/selenium_test.py", line 25, in setUp
    super(SeleniumTest, self).setUp()
  File "/home/nizarmah/Documents/Work/opencraft/xblock/xblock-skytap/.tox/py38-django30/lib/python3.8/site-packages/bok_choy/web_app_test.py", line 121, in setUp
    self.browser = browser(tags, self.proxy)
  File "/home/nizarmah/Documents/Work/opencraft/xblock/xblock-skytap/.tox/py38-django30/lib/python3.8/site-packages/bok_choy/browser.py", line 247, in browser
    browser_instance = Promise(
  File "/home/nizarmah/Documents/Work/opencraft/xblock/xblock-skytap/.tox/py38-django30/lib/python3.8/site-packages/bok_choy/promise.py", line 106, in fulfill
    raise BrokenPromise(self)
bok_choy.promise.BrokenPromise: Promise not satisfied: Browser is instantiated successfully.
----------------------------------------------------------------------
```

~~I do not understand how to fix it. I did see the following line mentioned in the repository [edx/bok-choy](https://github.com/edx/bok-choy).~~
> On Ubuntu Linux 18.04 you might have to install firefox-geckodriver (for Firefox) and/or chromium-chomedriver (for Chromium), especially if you hit the following Error when running bok_choy:
> ```
> bok_choy.promise.BrokenPromise: Promise not satisfied: Browser is instantiated successfully.
> ```

~~I did in fact add `chromedriver` to my `/usr/local/bin` and I even renamed it to `chromium-chromedriver` and still no luck. Please let me know.~~

#

:tada: I fixed it. I finally understand exactly what I'm doing with `tox` I guess. So the issue was first that the chrome options wasn't working, this was caused by the `bok-choy` and `selenium` outdated libraries. Then, there was an issue with `passenv` where not all the variables were being passed onto `tox`.

--- a/circle.yml
+++ b/circle.yml
@@ -5,16 +5,22 @@ jobs:
       - image: circleci/python:3.8.0
     steps:
       - checkout
-      - restore_cache:
-          key: dependencies-{{ checksum "circle.yml" }}-{{ checksum "requirements.txt" }}-{{ checksum "test-requirements.txt" }}
+      - run:
+          name: Install Python 3.5
+          command: |
+            sudo apt-get update
+            sudo apt-get install build-essential libreadline-gplv2-dev libncursesw5-dev libssl-dev libsqlite3-dev tk-dev libgdbm-dev libc6-dev libbz2-dev
+            cd /usr/src
+            sudo wget https://www.python.org/ftp/python/3.5.9/Python-3.5.9.tgz
+            sudo tar xzf Python-3.5.9.tgz
+            cd Python-3.5.9
+            sudo ./configure --enable-optimizations
+            sudo make altinstall
+            cd ~
       - run:
           name: Install tox
           command: |
-            pip install tox
-      - save_cache:
-          key: dependencies-{{ checksum "circle.yml" }}-{{ checksum "requirements.txt" }}-{{ checksum "test-requirements.txt" }}
-          paths:
-            - "venv"
+            sudo pip install tox
       - run:
           name: Run tests
           command: |

--- a/tox.ini
+++ b/tox.ini
@@ -10,15 +10,23 @@ envlist = py{35}-django{22}, py{38}-django{22,30}, py38-quality
 setenv =
     SELENIUM_BROWSER=chrome
 
+passenv =
+    *
+
 deps =
     -rtest-requirements.txt
     django22: Django==2.2.13
     django30: Django>=3.0,<3.1
 
-commands =
+commands_pre =
     pip install -r {envdir}/src/xblock-sdk/requirements/base.txt
     pip install -r {envdir}/src/xblock-sdk/requirements/test.txt
-    
+
+    # upgrade both bok-choy and selenium to avoid BrokenPromise Error
+    # caused due to deprecated chromeOptions issue
+    pip install selenium==3.141.0 bok-choy==1.1.1
+
+commands =
     python run_tests.py
 
 [testenv:py38-quality]

--- a/tox.ini
+++ b/tox.ini
@@ -6,7 +6,7 @@
 [tox]
 envlist = py{35}-django{22}, py{38}-django{22,30}, py38-quality
 
-[testenv]
+[testenv:pyNM-djangoNM]
 setenv =
     SELENIUM_BROWSER=chrome
 

